### ISSUE TITLE
Data should be able to take in StringIO

### ISSUE THREAD

**Issue by JGSweets** [capitalone/DataProfiler#197]
**Is your feature request related to a problem? Please describe.**
Yes, cannot ingest StringIO or BytesIO objects.

**Describe the outcome you'd like:**
Should fail with our reader as such (IF we took in StringIO, need to create this data as a file):
```python
import dataprofiler as dp
from io import StringIO
s = """
1


4"""
data = dp.Data(s)
```

**Comment by JGSweets** [comment]
partially being addressed in #292

**Comment by JGSweets** [comment]
This is now live.
